.github/workflows/release.yml

```diff
@@ -52,30 +70,75 @@ jobs:
         run: npm config set //registry.npmjs.org/:_authToken=${{ secrets.DIALTONE_NPM_TOKEN }}
 
       # note that pnpm run build is executed before publishing via prepublishOnly
-      - name: Deploy production
+      - name: Publish production
         if: ${{ github.ref == 'refs/heads/production' }}
         run: pnpm nx run-many --target=release
 
-      - name: Deploy prerelease
+      - name: Publish pre-release
         if: ${{ github.ref != 'refs/heads/production' }}
-        run: pnpm nx run-many --target=release --args="--tag ${{ steps.branch.outputs.BRANCH }}"
+        run: pnpm nx run-many --target=release --args="--tag ${{ needs.get-branch-name.outputs.current_branch }}"
 
-      - name: Set registry to github packages
+      - name: Set registry to GitHub packages
         run: npm config set @dialpad:registry https://npm.pkg.github.com
 
-      - name: Set github auth token
+      - name: Set GitHub auth token
         run: npm config set //npm.pkg.github.com/:_authToken=${{ secrets.GITHUB_TOKEN }}
 
       # note that pnpm run build is executed before publishing via prepublishOnly
```

Comment:
Hmm seems there is a `prepublishOnly: "pnpm run build"` in each individual package. Does this still make sense? For one doesn't this run the entire `pnpm run build` for each package? Should it be moved to the root package, and then also have a single command to publish in the root package rather than using run-many? If not, should we be running the build for each package individually instead of the whole build?

Reply:
It makes sense to run the build before publishing but we might want to move that config to a more centralized config, I'll move it to `nx.json` on targetDefaults to let NX now that release depends on building dependencies so it can optimize it.

We can do it in two ways:
- Using NX run-many 
- Using pnpm -r publish

Using NX run-many should be the way to go as NX would use cache and optimize the build and release order.

Reply:
Sure the nx run-many sounds like the way to go then.

I find it a bit confusing when we should use nx commands vs pnpm commands.

Reply:
Yes, it's a bit confusing, I'm reading more about to know exactly when we need to use pnpm, but NX should be the way to go in most cases AFAIK for now.